ui/packages/ui/src/main/webapp/js/views/installer/IdpConfiguration.view.js

```diff
@@ -0,0 +1,689 @@
+      // default arguments
+      value = value || ''
+      isInitialization = isInitialization || false
+
+      if (typeof value === 'object') {
+        // if the given value is an event, set the new value to ''
+        value = ''
+      }
+
+      var entry = new DeletableEntry({
+        name: name,
+        value: value,
+        type: this.type,
+      })
+      entry.render()
+
+      this.entriesElement[0].append(entry.el)
+
+      if (!isInitialization) {
+        this.values.push(value)
+        wreqr.vent.trigger('idpConfigModified')
+      }
+    },
+    updateValue: function(event) {
+      var target = event.currentTarget
+      var entry = target.parentElement
+      var value = target.value
+      var oldValue = target.getAttribute('oldValue')
+
+      this.values[this.values.indexOf(oldValue)] = value
+
+      target.setAttribute('oldValue', value)
+
+      var isValid = validateValue(this.type, value)
+
+      wreqr.vent.trigger('idpConfigModified')
+    },
+    removeValue: function(event) {
+      var removeButton = event.currentTarget
+      var entry = removeButton.parentElement
+      var inputField = entry.firstChild
+      var inputValue = inputField.value
+
+      this.values.splice(this.values.indexOf(inputValue), 1)
+      this.entriesElement[0].removeChild(entry)
+
+      wreqr.vent.trigger('idpConfigModified')
+    },
+    getConfig: function() {
+      return {
+        id: this.id,
+        value: this.values,
+      }
+    },
+    hasErrors: function() {
+      return !validateValues(this.type, this.values)
+    },
+  })
+
+  var DeletableEntry = Marionette.ItemView.extend({
+    template: deletableEntry,
+    tagName: 'div',
+    className: 'deletable-entry',
+    events: {
+      'change .form-data': 'checkAndSetValid',
+    },
+    initialize: function(options) {
+      this.name = options.name
+      this.value = options.value
+      this.type = options.type
+
+      this.typeName = getTypeNameFromType(this.type)
+      this.isValid = validateValue(this.type, this.value)
+    },
+    serializeData: function() {
+      return {
+        value: this.value,
+        typeName: this.typeName,
+      }
+    },
+    onRender: function() {
+      this.errorElement = this.$el.find('.error-message')
+
+      if (this.isValid) {
+        this.hideError()
+      } else {
+        this.showError()
+      }
+    },
+    checkAndSetValid: function(event) {
+      this.value = event.currentTarget.value
+      this.isValid = validateValue(this.type, this.value)
+
+      if (this.isValid) {
+        this.hideError()
+      } else {
+        this.showError()
+      }
+    },
+    hideError: function() {
+      this.errorElement[0].setAttribute('hidden', '')
+    },
+    showError: function() {
+      this.errorElement[0].removeAttribute('hidden')
+    },
+    hasErrors: function() {
+      return !this.isValid
+    },
+  })
+
+  var IdpConfigurationTable = Marionette.Layout.extend({
```

Comment:
📘 Responsible for creating and displaying different types of `IdpConfigurationEntry`s.